### PR TITLE
add cli docs

### PR DESCRIPTION
**Pull request by Bishoy-at-pieces**

### CODE REVIEW

File: docs/extensions-plugins/cli.mdx [hunk 1]
```diff
@@ -0,0 +1,395 @@
+---
+title: Pieces for Developers CLI Tool
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:2]: I think in general, just a look at proper formatting should be followed. 

Also we should add some assets - GIFs of using the CLI.

File: docs/extensions-plugins/cli.mdx [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+You can use `.` or `/` to refer to the current directory 
+
+```bash
+pieces ask "your question" -f /file1 /file2 folder -s 1 2 3
+```
+
+
+
+
+##### Commiting to github
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:267]: Github

Comment (shivay-at-pieces) [reply docs/extensions-plugins/cli.mdx:267]: Also you can keep the title as Auto Commit to GIthub

File: docs/extensions-plugins/cli.mdx [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+Auto commit the code to github and generate a commit message you can use the `-p` or `--push` flags to push the code to the repo too.
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:269]: I think this description should be updated. We should also add some additional details on how it works, ie.e looking at the git diff and then using that to give you a good looking commit message

File: docs/extensions-plugins/cli.mdx [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+
+##### change the llm model you are using:
+
+Change the model in the ask command.
+
+```bash
+pieces change_model [MODEL_INDEX]
+```
+##### Conversations command
+###### Coversations
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:205]: "Coversations" should be "Conversations"

File: docs/extensions-plugins/cli.mdx [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+You can create a new conversation that will be used in the ask command.
+
+```bash
+pieces conversation -n
+```
+
+Rename a conversation
+
+```bash
+pieces conversation -r My awsome name
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:230]: "awsome" should be "awesome" in the conversation rename example

File: docs/extensions-plugins/cli.mdx [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+##### Open an asset:
+
+Opens an asset from a list or search. If only "open" is used then it will open your most recent asset. This also creates a link to the asset's code.
+
+```bash
+pieces open [ITEM_INDEX]
+```
+##### Save, Create, Edit, and Delete Assets
+
+The save create edit and delete commands currrently work on the current asset which is by defualt set to the most recent one and you can change the current asset to anything using the open command above.
```
Comment (shivay-at-pieces) [inline docs/extensions-plugins/cli.mdx:117]: currently

File: docs/extensions-plugins/cli.mdx (hunk 1 above)
Comment (shivay-at-pieces) [reply docs/extensions-plugins/cli.mdx:2]: Also please remove the Inconsistent capitalization: sometimes you use "Pieces OS" and other times "Pieces os"

And you can remove all the typos / grammatical errors

Comment (shivay-at-pieces) [reply docs/extensions-plugins/cli.mdx:2]: Also if you can follow the structure as recommended by @mason-at-pieces for other products

**Review by shivay-at-pieces — CHANGES_REQUESTED** [review]

**Comment by cole-at-pieces** [comment]
"Pieces CLI Tool" should be proper naming convention. We can still mention it's a Python CLI tool, but any proper noun should be our name for it.

**Comment by Bishoy-at-pieces** [comment]
I think I will stick with Pieces CLI tool 
should I capitalize the "t" in tool?

**Comment by cole-at-pieces** [comment]
> I think I will stick with Pieces CLI tool should I capitalize the "t" in tool?

Yes I'd do that

**Comment by Bishoy-at-pieces** [comment]
Thank you @cole-at-pieces 
https://documentation-git-add-cli-agent-docs-pieces-for-developers.vercel.app/
check it and let me know if you have any other concerns 
everything should be to Pieces CLI Tool

**Comment by mason-at-pieces** [comment]
@Bishoy-at-pieces lets make sure that we are using h2's for subheadings. Seems like we skipped from h1 straight to h3

**Pull request merged**